ERCS/erc-7780.md

```diff
@@ -0,0 +1,178 @@
+
+Policy modules can be used to check what a UserOperation or action is trying to achieve and determine if this is allowed. Signer modules can be used to validate signatures on provided hashes. Stateless validators are modules that are used to both validate signatures and compare them to a calldata-provided data blob which could, for example, include owners to check signatures against.
+
+## Motivation
+
+The modules introduced by this proposal aim to create more composability around signature and permission verification.
+
+Policy and signer modules allow an account to make direct use of such a permissioning logic rather than relying on external modules to handle this. This has the upside of lower gas cost but the downside of less flexibility for users and developers that use the account.
+
+Stateless validators enable further composability around signature validation logic. In many cases, it does not make sense to re-write signature validation for new validators, but instead to use the existing ones. However, this is usually not possible since the validators rely on a stored configuration indexed by the `msg.sender`, which is expected to be an account. Stateless validators solve this problem by not relying on state to compare signature verficiation against, but instead to compare it against a calldata-provided argument.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+This standard introduces three new module types on top of the existing modules introduced by ERC-7579:
+
+- Policy (type id: 5)
+- Signer (type id: 6)
+- Stateless Validator (type id: 7)
+
+Note: A single module can be of multiple types.
+
+### Policy
+
+Policies MUST implement the `IModule` and the `IPolicy` interface and have module type id: `5`.
+
+```solidity
+interface IPolicy is IModule {
+
+		/**
+		 * Checks a userOp to determine if it should be executed
+		 * @param id The id of the policy
+		 * @param userOp The user operation to check
+		 * @return The validation data to return to the EntryPoint
+		 */
+	  function checkUserOpPolicy(
+			  bytes32 id,
+			  PackedUserOperation calldata userOp
+		)
+        external
+        payable
+        virtual
+        returns (uint256);
+
+		/**
+		 * Checks a signature to determine if it should be executed
+		 * @param id The id of the policy
+		 * @param sender The sender of the transaction
+		 * @param hash The hash of the transaction
+		 * @param sig The signature of the transaction
+		 * @return The validation data to return to the EntryPoint
+		 */
+		function checkSignaturePolicy(
```

Comment:
You might want to add some requirements around what this function does. Maybe something like:

> Policies MUST validate the given `hash` against `sig`. Policies MUST revert if the signature is invalid. Policies SHOULD verify that the sender is..."

I'm far from an expert, so obviously write requirements that make sense and don't just take my suggestion literally.